### PR TITLE
Built-in, throttled, and configurable retry support

### PR DESCRIPTION
**Pull request by sullivanmatt**
## Problem
This library is currently fire-once and forget, which may be undesirable.

## Solution
Added a Retry adapter with user-configurable parameters.  This adapter is throttled with an exponential back-off.  Out of the box, this mechanism will default to 5 attempts over a period of 1 minute.

### CODE REVIEW

**Comment by sullivanmatt** [comment]
@aetherdeity for your review as well

**Comment by AetherDeity** [comment]
tests failing

**Comment by sullivanmatt** [comment]
@zach-taylor I will squash-and-merge when this has been reviewed

File: splunk_handler/__init__.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 from threading import Timer
 
 import requests
+from requests.packages.urllib3.util.retry import Retry
+from requests.packages.urllib3 import PoolManager
```
Comment (zach-taylor) [inline splunk_handler/__init__.py]: Is this used anywhere?

**Comment by AetherDeity** [comment]
sure! +1

File: splunk_handler/__init__.py (hunk 1 above)
Comment (sullivanmatt) [inline splunk_handler/__init__.py]: @zach-taylor not anymore :trollface:

**Comment by sullivanmatt** [comment]
@zach-taylor release tagged

**Pull request merged**